storage/xtradb/handler/ha_innodb.cc

```diff
@@ -11402,6 +11487,17 @@ create_index(
 
 	ut_ad(key->flags & HA_FULLTEXT || !(index->type & DICT_FTS));
 
+	if(key->algorithm == HA_KEY_ALG_HASH){	
+		res = static_cast<char*>(mem_heap_alloc(index->heap, strlen("HASH_COL_") +
+			strlen(key->name) + 1));
```

Comment:
This is not necessary, but you could define a variable to contain result of strlen("HASH_COL_") and use that every place not call strlen every time.